templates/dart/lib/src/chunked_upload_io.dart.twig

```diff
@@ -32,14 +34,28 @@ Future<Response> chunkedUpload({
       headers: headers,
     );
   }
-  // read chunk and upload each chunk
+
   var offset = 0;
+  if(idParamName.isNotEmpty && params[idParamName] != 'unique()') {
+    //make a request to check if a file already exists
+    try {
+      res = await client.call(
+        HttpMethod.get,
+        path: path + '/' + params[idParamName],
+        headers: headers,
+      );
+      final int chunksUploaded = res.data['chunksUploaded'] as int;
+      offset = min(size, chunksUploaded * Client.CHUNK_SIZE);
```

Comment:
What happens if file is 100% uploaded and they try to re-upload something? We should probably instantly return the file.

Reply:
yes, it returns the file. Here I'm not sure what we should do, throw duplicate error like it happens for files < 5MB or just return the file. For now it returns the file.